link_shortener/form_routes.py

```diff
@@ -61,6 +112,7 @@ async def create_link_save(request, user):
                     identifier=str(uuid.uuid1()),
                     owner=user.email,
                     owner_id=user.id,
+                    password=form.password.data,
```

Comment:
Only applifters can manipulate with passwords, but I should do it anyway. Would pbkdf2_hmac using sha256 and a 32-byte salt (stored as an environment variable) be suitable for this?

Reply:
Well, AFAIK, the reason why the password should be stored encrypted (aka hashed) in a DB is that if the DB leaks, the attacker won't have an easy access to the passwords. In this case, the leaking the passwords is probably not so dangerous since the URLs protected by these passwords are probably not going to be anything sensitive and the passwords protecting them as well, but I think that it's better to be safe here than sorry later...

Well, yes, I think that any of the algorithms in the standard library is a good choice. Django seems to default `pbkdf2_hmac`. `scrypt` seems to be believed to be superior but meh... So yes, feel free to go for the solution you have suggested...

Reply:
Done:

- Created a new table for storing 🧂 
- Assigned an individual 🧂  to each link's identifier
- Authorisation now pulls the 🧂 corresponding to the authorised link's identifier, hashes the password input from the password form, and compares it to the already hashed password from the links database

To do:

- Copy the process of creating 🧂 to the create link form
- Update the link edit form to comply with password hashing

Reply:
All done